commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/PurePursuitControllers.scala

```diff
@@ -165,5 +163,70 @@ trait PurePursuitControllers {
       }
     )
   }
+
+  def followWayPointsController(wayPoints: Seq[Point], position: PeriodicSignal[Point])
+                               (implicit hardware: UnicycleHardware,
+                                signal: Signal[UnicycleProperties]): (PeriodicSignal[UnicycleSignal], Signal[Option[Length]]) = {
+    var previousLookAheadPoint: Option[Point] = None
+
+    val biSegmentPaths = wayPoints.sliding(3).map { points =>
+      (
+        Segment(points(0), points(1)),
+        if (points.size == 3) {
+          Some(Segment(points(1), points(2)))
+        } else {
+          None
+        }
+      )
+    }
+
+    var currPath: (Segment, Option[Segment]) = (
+      Segment(wayPoints(0), wayPoints(1)),
```

Comment:
Doesn't handle case of only one way point. Prepend initial position to fix this.